modmail/extensions/extension_manager.py

```diff
@@ -0,0 +1,301 @@
+            extensions = sorted(
+                ext
+                for ext in self.bot.extensions.keys() & self.all_extensions
+                if ext not in (self.get_black_listed_extensions())
+            )
+
+        await ctx.send(self.batch_manage(Action.UNLOAD, *extensions))
+
+    @extensions_group.command(name="reload", aliases=("r", "rl"))
+    async def reload_extensions(self, ctx: Context, *extensions: ExtensionConverter) -> None:
+        """
+        Reload extensions given their fully qualified or unqualified names.
+
+        If an extension fails to be reloaded, it will be rolled-back to the prior working state.
+
+        If '\*' or '\*\*' is given as the name, all currently loaded extensions will be reloaded.
+        """  # noqa: W605
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+
+        if "*" in extensions or "**" in extensions:
+            extensions = self.bot.extensions.keys() & self.all_extensions.keys()
+
+        await ctx.send(self.batch_manage(Action.RELOAD, *extensions))
+
+    @extensions_group.command(name="list", aliases=("all", "ls"))
+    async def list_extensions(self, ctx: Context) -> None:
+        """
+        Get a list of all extensions, including their loaded status.
+
+        Red indicates that the extension is unloaded.
+        Green indicates that the extension is currently loaded.
+        """
+        embed = Embed(colour=Colour.blurple())
+        embed.set_author(
+            name=f"{self.type.capitalize()} List",
+        )
+
+        lines = []
+        categories = self.group_extension_statuses()
+        for category, extensions in sorted(categories.items()):
+            # Treat each category as a single line by concatenating everything.
+            # This ensures the paginator will not cut off a page in the middle of a category.
+            category = category.replace("_", " ").title()
+            extensions = "\n".join(sorted(extensions))
+            lines.append(f"**{category}**\n{extensions}\n")
+
+        log.debug(f"{ctx.author} requested a list of all {self.type}s. " "Returning a paginated list.")
+
+        # TODO: since we currently don't have a paginator.
+        await ctx.send("".join(lines) or f"There are no {self.type}s installed.")
+
+    @extensions_group.command(name="refresh", aliases=("rewalk",))
+    async def rewalk_extensions(self, ctx: Context) -> None:
+        """
+        Refreshes the list of extensions.
+
+        Typical use case is in the event that the existing extensions have changed while the bot is running.
+        """
+        log.debug(f"Refreshing list of {self.type}s.")
+
+        # make sure the new walk contains all currently loaded extensions, so they can be unloaded
+        loaded_extensions = dict()
+        for ext in self.all_extensions.items():
+            if ext in self.bot.extensions:
+                loaded_extensions.update(ext)
```

Comment:
This always fails. `.items()` returns a typle, `bot.extensions` is a dictionary of `name: extension`. A tuple will never be in a list like that, so this will never treat any extension as loaded. Instead, we should just use nothing, I believe, as that ought be more appropriate. 

```py
for ext_name, should_load in self.all_extensions.items():
    if ext_name in self.bot.extensions:
        loaded_extensions[ext_name] = should_load
```

Also.. is it possible for us to write this as a generator expression passing it into loaded_extensions?